test/jit/test_dataclasses.py

```diff
@@ -0,0 +1,107 @@
+# Owner(s): ["oncall: jit"]
+
+from dataclasses import dataclass, field, InitVar
+from hypothesis import given, strategies as st
+from torch.testing._internal.jit_utils import JitTestCase
+from typing import List, Optional
+import sys
+import torch
+import unittest
+
+if __name__ == '__main__':
+    raise RuntimeError("This test file is not meant to be run directly, use:\n\n"
+                       "\tpython test/test_jit.py TESTNAME\n\n"
+                       "instead.")
+
+# Example jittable dataclass
+@dataclass(order=True)
+class Point:
+    x: float
+    y: float
+    norm: Optional[torch.Tensor] = None
+
+    def __post_init__(self):
+        self.norm = (torch.tensor(self.x) ** 2 + torch.tensor(self.y) ** 2) ** 0.5
+
+
+# Hypothesis strategies
+ExtendedReals = st.floats(allow_infinity=True, allow_nan=False)
+
+class TestDataclasses(JitTestCase):
+    # We only support InitVar in JIT dataclasses for Python 3.8+ because it would be very hard
+    # to support without the `type` attribute on InitVar (see comment in _dataclass_impls.py).
+    @unittest.skipIf(sys.version_info < (3, 8), "InitVar not supported in Python < 3.8")
+    def test_init_vars(self):
+        @dataclass(order=True)
+        class Point2:
+            x: float
+            y: float
+            norm_p: InitVar[int] = 2
+            norm: Optional[torch.Tensor] = None
+
+            def __post_init__(self, norm_p: int):
+                self.norm = (torch.tensor(self.x) ** norm_p + torch.tensor(self.y) ** norm_p) ** (1 / norm_p)
```

Comment:
I got this error: 
```
RuntimeError: The following operation failed in the TorchScript interpreter.
Traceback of TorchScript (most recent call last):
  File "<string>", line 2, in fn
def fn(x: float, y: float):
    pt = Point(x, y)
         ~~~~~ <--- HERE
    return pt.norm
  self.y = y
  self.norm = norm
  self.__post_init__()
  ~~~~~~~~~~~~~~~~~~ <--- HERE
  File "/data/sandcastle/boxes/eden-trunk-hg-fbcode-fbsource/fbcode/buck-out/opt/gen/caffe2/test/jit#binary,link-tree/jit/test_dataclasses.py", line 24, in __post_init__
    def __post_init__(self):
        self.norm = (torch.tensor(self.x) ** 2 + torch.tensor(self.y) ** 2) ** 0.5
                                                 ~~~~~~~~~~~~ <--- HERE
RuntimeError: value cannot be converted to type float without overflow
```
running with internal toolchain.  Possibly because sometimes hypothesis fed in numbers slightly too large.

Reply:
Oh that makes sense. I can change the Hypothesis strategy to be more restrictive later tonight. Also if you want to go ahead and fix it feel free to modify the PR.

Reply:
Just made the Hypothesis strategy bounded between -1e4 and 1e4, hopefully that fixes it.